### ISSUE TITLE
 Handle the "spatial" attribute in onnx BatchNormalization

### ISSUE THREAD

**Issue by oleid** [snipsco/tract#257]
I tried Mobilenet v2 example from ONNX model zoo to check if it works, given that the corresponding tensorflow model is part of the examples.

https://s3.amazonaws.com/onnx-model-zoo/mobilenet/mobilenetv2-1.0/mobilenetv2-1.0.onnx

Turned out that it crashes on start up. Traced it back an assert assuring `spatial = 0`
```
tract_onnx::ops::nn::batch_normalization::h9eb0af9dc679fb50 mod.rs:117
tract_onnx::model::ParsingContext::parse_graph::haa33f85c9f8fbf5e model.rs:108
tract_onnx::model::Onnx::parse::h5ff6da06eaeb2b66 model.rs:208
_$LT$tract_onnx..model..Onnx$u20$as$u20$tract_hir..framework..Framework$LT$tract_onnx..pb..ModelProto$GT$$GT$::model_for_proto_model::hf9a44d34d9c02e91 model.rs:221
tract_hir::framework::Framework::model_for_read::h95a5740e42f84c36 framework.rs:32
tract_hir::framework::Framework::model_for_path::h68896d9cea1dd51a framework.rs:39
[...]
```

For pytorch, supporting _spatial_ was trivial (just ignore the attribute):

https://github.com/pytorch/pytorch/pull/9492/files

Would it be similar easy for tract?

To check that, I removed the assert. But that leads to node analysis errors. For reference:

```Error: TractError(Msg("Failed analyse for node #268 \"mobilenetv20_features_conv0_fwd\" Conv"), State { next_error: Some(TractError(Msg("Infering facts"), State { next_error: Some(TractError(Msg("Applying rule inputs[0].shape[1] == 1*{inputs[1].shape[1]}: Impossible to unify Val(224) with Val(3)."), State { next_error: None, backtrace: InternalBacktrace { backtrace: Some(stack backtrace: [..]```

**Comment by kali** [comment]
Hey !

This model is for a relatively old version of ONNX (operator set 7).

tract onnx support was started for ONNX operator set version 9 and we make efforts for making adjustments to follow the newer versions of ONNX when issues arrise. Note that we do not make check for pre-9 version, maybe we should.

In the case of BatchNormalization, support for the `spatial` attribute has been removed with onnx operator set 9. It would be possible to add it while maintaining compatibility (we already have had to deal with operators that change their behaviour between opset 9 and opset 10, so we can support this logic), but we may run into several other discrepancies between opset 7 and opset 9 for other operators.

If you have a specific interest in mobilenet, we have an example running it in its tensorflow form. Also this project https://github.com/onnx/tensorflow-onnx (a converter tf->onnx) apparently uses it in their test suites so it should work (and they can target various onnx operator sets).

**Comment by oleid** [comment]
Ah, discrepancies due to an old version makes sense. Yes, I think at least warning about an unsupported version would make sense. You can always change the version during conversion. AFAIR tf2onnx defaults to version 8 at the moment. So chances are high to create nets with the wrong version.

**Comment by kali** [comment]
I added a warning and debug message about operator set on master. It will be part of the next release.

**Comment by oleid** [comment]
Thanks a lot! :)
